Shared/Models/Device.swift

```diff
@@ -0,0 +1,99 @@
+    let deleted: Bool
+    let deletedAt: String?
+    let removed: Bool
+    let removedAt: String?
+    let createdAt: String
+    let updatedAt: String
+    let userId: Int?
+    let parent_id: String?
+    var hasBackup: Bool = false
+
+    var isCurrentDevice: Bool {
+        return ConfigLoader().getDeviceName() == self.plain_name
+    }
+
+    init(id: Int, uuid: String, parentId: String?, parentUuid: String?, name: String?, plain_name: String? = nil, bucket: String?, user_id: Int?, encrypt_version: String?, deleted: Bool, deletedAt: String?, removed: Bool, removedAt: String?, createdAt: String, updatedAt: String, userId: Int?, parent_id: String?, hasBackup: Bool) {
+        let decrypt = Decrypt()
+
+        self.id = id
+        self.uuid = uuid
+        self.parentId = parentId
+        self.parentUuid = parentUuid
+        self.name = name
+        if let plainName = plain_name {
+            self.plain_name = plainName
+        } else {
+            do {
+                self.plain_name = try decrypt.decrypt(base64String: name ?? "", password: "\(DecryptUtils().getDecryptPassword(bucketId: bucket ?? ""))")
+            } catch {
+                error.reportToSentry()
+                self.plain_name = ""
+            }
+        }
+        self.bucket = bucket
+        self.user_id = user_id
+        self.encrypt_version = encrypt_version
+        self.deleted = deleted
+        self.deletedAt = deletedAt
+        self.removed = removed
+        self.removedAt = removedAt
+        self.createdAt = createdAt
+        self.updatedAt = updatedAt
+        self.userId = userId
+        self.parent_id = parent_id
+        self.hasBackup = hasBackup
+    }
+
+    init(from deviceAsFolder: DeviceAsFolder) {
+        let decrypt = Decrypt()
+
+        self.id = deviceAsFolder.id
+        self.uuid = deviceAsFolder.uuid
+        self.parentId = deviceAsFolder.parentId
+        self.parentUuid = deviceAsFolder.parentUuid
+        self.name = deviceAsFolder.name
+        if let plainName = deviceAsFolder.plain_name {
+            self.plain_name = plainName
+        } else {
+            do {
+                self.plain_name = try decrypt.decrypt(base64String: deviceAsFolder.name ?? "", password: "\(DecryptUtils().getDecryptPassword(bucketId: deviceAsFolder.bucket ?? ""))")
+            } catch {
+                error.reportToSentry()
+                self.plain_name = ""
+            }
+        }
+        self.bucket = deviceAsFolder.bucket
+        self.user_id = deviceAsFolder.user_id
+        self.encrypt_version = deviceAsFolder.encrypt_version
+        self.deleted = deviceAsFolder.deleted
+        self.deletedAt = deviceAsFolder.deletedAt
+        self.removed = deviceAsFolder.removed
+        self.removedAt = deviceAsFolder.removedAt
+        self.createdAt = deviceAsFolder.createdAt
+        self.updatedAt = deviceAsFolder.updatedAt
+        self.userId = deviceAsFolder.userId
+        self.parent_id = deviceAsFolder.parent_id
```

Comment:
just a tip, we don't use snake casing in the project, but the API returns us properties in snake casing, to maintain consistency in the codebase, and avoid coupling with the backend naming, I suggest you to map the API response properties like this:

`self.parentId = deviceAsFolder.parent_id`

so you are consistent with the variable casing in the codebase